src/pages/Invite/Invite.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link as RouterLink, Redirect, useParams } from 'react-router-dom';
+import { useMutation, useQuery } from '@apollo/client';
+import { Box, Button, ButtonGroup, Center, Heading, Link, Spinner, Text, VStack, useToast } from '@chakra-ui/react';
+
+import Toast from 'components/dashboard/Toast/Toast';
+
+import { UPDATE_USERS_TEAM } from 'data/actions/type';
+import { GET_TEAM_INFO } from 'data/gql/team';
+import { JOIN_TEAM } from 'data/gql/user';
+
+const InviteLayout = props => (
+  <VStack spacing={4} align="left">
+    <Box>
+      <Link color="gray.500" mb="8px" as={RouterLink} to="/">
+        Back to home
+      </Link>
+      <Heading size="h1" as="h1">
+        Team Invite
+      </Heading>
+    </Box>
+    {props.children}
+  </VStack>
+);
+
+function Invite(props) {
+  const dispatch = useDispatch();
+  const toast = useToast();
+  const [shouldRedirect, setShouldRedirect] = useState(false);
+
+  const { teamId } = useParams();
+  const { userId } = useSelector(state => state.auth.user);
+
+  const [joinTeam, { loading: joinTeamLoading, data: joinTeamData }] = useMutation(JOIN_TEAM);
```

Comment:
notice that `useMutation` returns `loading` and `data` just like `useQuery`. You previously put in your own loading state with `useState`, but no need!